src/modulefinder/sentry_modulefinder_apple.c

```diff
@@ -132,10 +132,25 @@ sentry__modules_get_list(void)
     sentry__mutex_lock(&g_mutex);
     if (!g_initialized) {
         g_modules = sentry_value_new_list();
+        // TODO: maybe use `_dyld_image_count` and `_dyld_get_image_header`?
+        // Those functions are documented to not be thread-safe, though using
+        // the `register_X` functions are also unsafe because they lack a
+        // corresponding `unregister` function, and will thus crash when sentry
+        // itself is unloaded.
```

Comment:
Does this even apply here? You're running inside a mutex lock, so thread safety shouldn't be a concern.

Reply:
I think its rather that you can `dlopen` in other threads, so the count/get-image get out of sync… per docs, get-image should gracefully return NULL on out-of-bounds, but with concurrent changes, there is a chance to miss some modules.

Reply:
I see, thanks for clarifying 👍